llvm/lib/Target/Xtensa/Disassembler/XtensaDisassembler.cpp

```diff
@@ -269,13 +316,31 @@ DecodeStatus XtensaDisassembler::getInstruction(MCInst &MI, uint64_t &Size,
                                                 ArrayRef<uint8_t> Bytes,
                                                 uint64_t Address,
                                                 raw_ostream &CS) const {
-  uint32_t Insn;
+  uint64_t Insn;
   DecodeStatus Result;
 
+  // Parse 16-bit instructions
+  if (hasDensity()) {
+    Result = readInstruction16(Bytes, Address, Size, Insn, IsLittleEndian);
+    if (Result == MCDisassembler::Fail)
+      return MCDisassembler::Fail;
+    LLVM_DEBUG(dbgs() << "Trying Xtensa 16-bit instruction table :\n");
+    Result = decodeInstruction(DecoderTable16, MI, Insn, Address, this, STI);
```

Comment:
In the disassembler test, make sure to test mismatches (e.g. !hasDensity and there are encoded values using the extension)

Reply:
Do you mean try to disassemble 16-bit code with the hasDensity option disabled and check that it produces erroneous decoding?

Reply:
Yes, it should not crash